components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceOrganizationProviderServiceImpl.java

```diff
@@ -0,0 +1,331 @@
+        for (DeviceOrganizationMetadataHolder tempHolder: tempDevicesInOrganization) {
+            if (tempHolder.getDeviceId().equals(deviceId)) {
+                indexInArray = tempDevicesInOrganization.indexOf(tempHolder);
+                return indexInArray;
+            }
+        }
+        return -1;
+    }
+
+    @Override
+    public List<DeviceOrganizationVisNode> generateNodes() {
+        List<DeviceOrganizationMetadataHolder> tempDevicesInOrganization = new ArrayList<>();
+        try {
+            tempDevicesInOrganization = this.getDevicesInOrganization();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error while getting devices in organization";
+            log.error(msg, e);
+        }
+        List<DeviceOrganizationVisNode> nodes = new ArrayList<>();
+        for (DeviceOrganizationMetadataHolder tempHolder: tempDevicesInOrganization) {
+            String tempId = tempHolder.getDeviceId();
+            String tempLabel = tempHolder.getDeviceName();
+            int tempSize = 21;
+            String tempColor = "#0000ff";
+            if (tempHolder.getDeviceId().equals("server")) {
+                tempColor = "#ffa500";
+                tempSize = 70;
+            }
+            if (!tempHolder.getDeviceId().equals("server") && tempHolder.getIsGateway() == 1) {
+                tempSize = 33;
+                tempColor = "#ff0000";
+            }
+            if (tempHolder.getState() == 1 || tempHolder.getState() == 0) {
+                tempColor = "#d3d3d3";
+            }
+            nodes.add(new DeviceOrganizationVisNode(tempId,tempLabel,tempSize,tempColor));
+        }
+        return nodes;
+    }
+
+    @Override
+    public List<DeviceOrganizationVisEdge> generateEdges() {
+        List<DeviceOrganizationMetadataHolder> tempDevicesInOrganization = new ArrayList<>();
+        try {
+            tempDevicesInOrganization = this.getDevicesInOrganization();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error while getting devices in organization";
+            log.error(msg, e);
+        }
+        List<DeviceOrganizationVisEdge> edges = new ArrayList<>();
+        for (DeviceOrganizationMetadataHolder tempHolder: tempDevicesInOrganization) {
+            String child = tempHolder.getDeviceId();
+            String parent = tempHolder.getParent();
+            if (parent.equals("") || parent.equals(" ")) {
+                continue;
+            } else {
+                edges.add(new DeviceOrganizationVisEdge(parent, child));
+            }
+        }
+        return edges;
+    }
+
+    @Override
+    public List<DeviceOrganizationNode> generateHierarchy() {
+        List<DeviceOrganizationNode> hierarchy = new ArrayList<>();
+        List<DeviceOrganizationNode> parentList = new ArrayList<>();
+        List<DeviceOrganizationNode> childList = new ArrayList<>();
+        hierarchy.add(new DeviceOrganizationNode("server"));
```

Comment:
Why can't we use a constant for "server" since it's used in several places?

Reply:
Fixed 47fa811b9bfdb72e1db01066273ffe2df5107a67